aiida_quantumespresso/workflows/matdyn/base.py

```diff
@@ -54,10 +52,9 @@ def report_error_handled(self, calculation, action):
         self.report('{}<{}> failed with exit status {}: {}'.format(*arguments))
         self.report('Action taken: {}'.format(action))
 
-
-@register_error_handler(MatdynBaseWorkChain, 600)
-def _handle_unrecoverable_failure(self, calculation):
-    """Handle calculations with an exit status below 400 which are unrecoverable, so abort the work chain."""
-    if calculation.exit_status < 400:
-        self.report_error_handled(calculation, 'unrecoverable error, aborting...')
-        return ErrorHandlerReport(True, True, self.exit_codes.ERROR_UNRECOVERABLE_FAILURE)
+    @process_handler(priority=600)
+    def handle_unrecoverable_failure(self, node):
+        """Handle calculations with an exit status below 400 which are unrecoverable, so abort the work chain."""
+        if node.is_failed and node.exit_status < 400:
+            self.report_error_handled(node, 'unrecoverable error, aborting...')
+            return ProcessHandlerReport(True, self.exit_codes.ERROR_UNRECOVERABLE_FAILURE)
```

Comment:
I think this would be a bit easier to understand if the parameters are passed by keyword (``do_break=True, exit_code=...``).

Reply:
In a sense yes, but since the `ProcessHandlerReport` is now the standard and shipped with `aiida-core`, it should be clear at some point. I will see how it looks, but merge this for now

Reply:
And thanks for the review! :+1: